source/common/upstream/upstream_impl.cc

```diff
@@ -1200,7 +1200,8 @@ ClusterImplBase::ClusterImplBase(
 
   if (info_->features() & ClusterInfoImpl::Features::HTTP3) {
 #if defined(ENVOY_ENABLE_QUIC)
-    if (cluster.transport_socket().name() != "envoy.transport_sockets.quic") {
+    if (cluster.transport_socket().DebugString().find("envoy.transport_sockets.quic") ==
```

Comment:
Out of curiosity, how does this differ from the existing code? In particular, looking at this change I don't understand how it "make(s) sure the proxy transport socket can be configured for auto_upstream with http/3", but I'm sure you do :)

Reply:
In the previous PR, we verified that for HTTP/3 we'd downgrade requests going through a proxy to TCP. We didn't actually have an e2e test for HTTP/3 with proxy config, becasue the HTTP/3 auto tests are all in the alt-svc extension directory, and the proxy tests are in the http_11_proxy directory, and thou shalt not mix and match extensions.  Unfortunately by not e2e testing we failed to verify that you could actually _configure_ Envoy with a http_11_proxy wrapping a quic transport socket.

Previously you couldn't because we checked for an explicit HTTP/3 socket.  This change allows the HTTP/3 socket to be wrapped by an outer http_11_proxy transport socket.